packages/ymax-planner/src/engine.ts

```diff
@@ -347,18 +381,74 @@ export const startEngine = async ({
 
   // TODO: verify consumption of paginated data.
   const { query } = signingSmartWalletKit;
-  const portfolioKeys = await query.vstorage.keys(VSTORAGE_PATH_PREFIX);
+  const portfolioKeys = await query.vstorage.keys(PORTFOLIOS_PATH_PREFIX);
   const portfolioKeyForDepositAddr = new Map() as Map<Bech32Address, string>;
   await makeWorkPool(portfolioKeys, undefined, async portfolioKey => {
     const status = await query.readPublished(
-      `${stripPrefix('published.', VSTORAGE_PATH_PREFIX)}.${portfolioKey}`,
+      `${stripPrefix('published.', PORTFOLIOS_PATH_PREFIX)}.${portfolioKey}`,
     );
     mustMatch(status, PortfolioStatusShapeExt, portfolioKey);
     const { depositAddress } = status;
     if (!depositAddress) return;
     portfolioKeyForDepositAddr.set(depositAddress, portfolioKey);
   }).done;
 
+  const subscriptionKeys = await query.vstorage.keys(PENDING_TXS_PATH_PREFIX);
+  console.warn(
+    `Found ${subscriptionKeys.length} existing subscriptions to monitor`,
+  );
+
+  const invitationMakersOffer = await getInvitationMakers(signingSmartWalletKit);
+  // Process existing pending subscriptions on startup
+  await makeWorkPool(subscriptionKeys, undefined, async subscriptionKey => {
+    const logIgnoredError = err => {
+      const msg = `Failed to process existing subscription: ${subscriptionKey}`;
+      console.error(msg, err);
+    };
+
+    const [vstorageSettlement] = await Promise.allSettled([
+      query.readPublished(
+        stripPrefix(
+          'published.',
+          `${PENDING_TXS_PATH_PREFIX}.${subscriptionKey}`,
+        ),
+      ),
+    ]);
+    if (vstorageSettlement.status === 'rejected') {
+      logIgnoredError(vstorageSettlement.reason);
+      return;
+    }
+
+    const subscriptionData = vstorageSettlement.value;
+    mustMatch(
+      subscriptionData,
+      BaseSubscriptionShape,
+      `subscription ${subscriptionKey}`,
+    );
```

Comment:
If we tolerating failures from `readPublished`, we might as well also tolerate bad shapes from it.
```suggestion
    if (!matches(subscriptionData, BaseSubscriptionShape)) {
      const err = assert.error(
        X`expected data ${subscriptionData} to match ${q(BaseSubscriptionShape)}`,
      );
      logIgnoredError(err);
      return;
    }
```